addons/arsenal/XEH_preInit.sqf

```diff
@@ -48,4 +48,58 @@ call FUNC(compileStats);
 // Setup Tools tab
 [keys (uiNamespace getVariable [QGVAR(configItemsTools), createHashMap]), LLSTRING(toolsTab), TOOLS_TAB_ICON, -1, true] call FUNC(addRightPanelButton);
 
+// TODO: make IDCs able to match IDX with simple math?
+GVAR(idxMap) = createHashMapFromArray [
+    [IDC_buttonPrimaryWeapon, IDX_VIRT_PRIMARY_WEAPONS],
+    [IDC_buttonHandgun, IDX_VIRT_HANDGUN_WEAPONS],
+    [IDC_buttonSecondaryWeapon, IDX_VIRT_SECONDARY_WEAPONS],
+    [IDC_buttonHeadgear, IDX_VIRT_HEADGEAR],
+    [IDC_buttonUniform, IDX_VIRT_UNIFORM],
+    [IDC_buttonVest, IDX_VIRT_VEST],
+    [IDC_buttonBackpack, IDX_VIRT_BACKPACK],
+    [IDC_buttonGoggles, IDX_VIRT_GOGGLES],
+    [IDC_buttonNVG, IDX_VIRT_NVG],
+    [IDC_buttonBinoculars, IDX_VIRT_BINO],
+    [IDC_buttonMap, IDX_VIRT_MAP],
+    [IDC_buttonGPS, IDX_VIRT_COMMS],
+    [IDC_buttonRadio, IDX_VIRT_RADIO],
+    [IDC_buttonCompass, IDX_VIRT_COMPASS],
+    [IDC_buttonWatch, IDX_VIRT_WATCH]
+];
+
+// Make new hashmaps for face/voice/insignia so mission makers can disable them
+// Merge instead of copying so we don't inherit final status
+GVAR(faceCache) = createHashMap;
+GVAR(faceCache) merge [uiNamespace getVariable QGVAR(faceCache), true];
+GVAR(voiceCache) = createHashMap;
+GVAR(voiceCache) merge [uiNamespace getVariable QGVAR(voiceCache), true];
+GVAR(insigniaCache) = createHashMap;
+GVAR(insigniaCache) merge [uiNamespace getVariable QGVAR(insigniaCache), true];
+
+{
+    private _varName = format [QGVAR(%1), _x];
+    SETMVAR(_varName,createHashMap);
+    GETMVAR(_varName,createHashMap) merge
+    (missionNamespace getVariable _varName) merge [uiNamespace getVariable _varName, true];
+} forEach ["voiceCache", "faceCache", "insigniaCache"];
```

Comment:
I haven't checked if the code below works or does what it's supposed to do.
```suggestion
{
    private _varName = format [QGVAR(%1), _x];
    missionNamespace setVariable [_varName, +(uiNamespace getVariable [_varName, createHashMap])];
} forEach ["voiceCache", "faceCache", "insigniaCache"];
```

Reply:
this will make the missionNamespace hashmap final and unable to be edited

Reply:
this code shouldn't be here, whoops